server/src/main/java/org/elasticsearch/common/xcontent/XContentHelper.java

```diff
@@ -575,4 +587,37 @@ public static void writeTo(StreamOutput out, XContentType xContentType) throws I
             out.writeVInt(xContentType.ordinal());
         }
     }
+
+    /**
+     * Construct {@link XContentFieldFilter} using given includes and excludes
+     * @param includes fields to keep, wildcard supported
+     * @param excludes fields to remove, wildcard supported
+     * @return filter using {@link XContentMapValues#filter(String[], String[])} if wildcard found in excludes, otherwise return filter using {@link XContentParser}
+     */
+    public static XContentFieldFilter newFieldFilter(String[] includes, String[] excludes) {
```

Comment:
Can this method live directly in XContentFieldFilter instead? That way it's available everywhere that the x-content lib is available.